interfaces/admin_cli.py

```diff
@@ -0,0 +1,501 @@
+        """
+        User returns to previous menu after action completed.
+        """
+        while True:
+            option = input("Enter R to return back to the previous menu.\n > ")
+            if option.upper() == 'R':
+                AdminShell(self.user).cmdloop()
+            else:
+                print("\033[31m {}\033[00m".format("** Invalid input. Enter R to return back to main menu."))
+                continue
+
+    # ----- basic commands for admins-----
+    def do_profile(self, arg):
+        """
+        Print user info
+        """
+        print("\n\033[100m\033[4m\033[1m{}\033[0m\n".format("Your details:") +
+              f'Username: {self.user.username}\n'
+              f'Role: {self.user.__class__.__name__}\n')
+        self.return_previous_page()
+
+    def do_manage_plan(self, arg):
+        """
+        Enter PlanMenu for further actions
+        """
+        PlanMenu(self.user).cmdloop()
+
+    def do_manage_volunteer_account(self, arg):
+        """
+        Enter ManageVolunteerMenu for further actions
+        """
+        ManageVolunteerMenu(self.user).cmdloop()
+
+    def do_manage_refugee_profile(self, arg):
+        """
+        Enter ManageCampMenu for further actions
+        """
+        ManageRefugeeMenu(self.user).cmdloop()
+
+    def do_exit(self, arg):
+        super().do_exit(self)
+
+
+class PlanMenu(AdminShell):
+    """
+    PlanMenu is the command line interface for emergency plan management.
+    """
+    plan_options = {"logout": 0, "create_plan": 1, "view_plan": 2,
+                    "close_plan": 3,
+                    "return_main_menu": "r"}
+
+    def plan_menu(self) -> None:
+        print("\033[100m\033[4m\033[1mManage Emergency Plan\033[0m\n"
+              "[ 1 ] Create a new emergency plan\n"
+              "[ 2 ] View an emergency plan\n"
+              "[ 3 ] Close an emergency plan\n\n"
+              "[ R ] Return to Main Menu\n")
+
+    def precmd(self, option: str) -> str:
+        """
+        Transfer option numbers to function name for Admin Menu
+        """
+        if option.isdigit() and int(option) in list(self.plan_options.values()):
+            return list(self.plan_options.keys())[list(self.plan_options.values()).index(int(option))]
+        elif option.upper() == 'X' or option.upper() == 'R':
+            return list(self.plan_options.keys())[list(self.plan_options.values()).index(option.lower())]
+        else:
+            return "invalid_input"
+
+    def preloop(self) -> None:
+        """
+        Display plan menu.
+        """
+        self.plan_menu()
+
+    def return_previous_page(self):
+        while True:
+            option = input("Enter R to return back to the previous menu.\n > ")
+            if option.upper() == 'R':
+                PlanMenu(self.user).cmdloop()
+            else:
+                print("\033[31m {}\033[00m".format("** Invalid input. Enter R to return back to main menu."))
+                continue
```

Comment:
Thanks for bringing this up and that's a good idea. I was thinking the user may want to stay on the same page for a while (to take down notes etc..) and then can choose when they want to return back to the menu so I created this function. If it is not necessary I'll remove it and return back to menu directly.

Reply:
Currently the method is overridden in each sub-menu, beacuse the users should go back to the sub-menu after their actions are completed instead of getting back to the main menu? This is just about the user flow design, and I can fix it to avoid complicating the flow!

Reply:
I mean we can directly redirect them back to the sub-menu, instead of prompting them to:
```
Enter R to return back to the previous menu.
```

Reply:
Fixed in new version